### PR TITLE
TW-136: add dynamic tokens whitelist

### PR DESCRIPTION
**Pull request by letier3110**

### CODE REVIEW

File: src/api.service.ts [hunk 1]
```diff
@@ -3,6 +3,7 @@ import axios from 'axios';
 import { TEMPLE_WALLET_API, TEMPLE_WALLET_EVERSTAKE_API_KEY, TEMPLE_WALLET_EXOLIX_API_KEY } from './utils/env.utils';
 import { isDcpNode } from './utils/network.utils';
 
+const WHITELIST_TOKENS_BASE_URL = 'https://raw.githubusercontent.com/madfish-solutions/tokens-whitelist/master/';
```
Comment (LendiHop) [inline src/api.service.ts:6]: redundant constant

**Review by LendiHop — APPROVED** [review]

File: src/utils/token-metadata.utils.ts [hunk 2]
```diff
@@ -97,6 +132,18 @@ export const withMetadataSlugs =
       ])
     );
 
+export const loadWhitelist$ = (selectedRpc: string): Observable<Array<TokenMetadataInterface>> =>
+  from(whitelistApi.get<WhitelistResponse>('tokens/quipuswap.whitelist.json')).pipe(
+    map(({ data }) =>
+      isDefined(data.tokens) && !isDcpNode(selectedRpc)
```
Comment (unixvb) [inline src/utils/token-metadata.utils.ts:138]: do not make request if this is DCP node

File: src/token/interfaces/account-token.interface.mock.ts [hunk 3]
```diff
@@ -1,8 +1,49 @@
 import { VisibilityEnum } from '../../enums/visibility.enum';
-import { MAINNET_TOKENS_METADATA } from '../data/tokens-metadata';
 import { getTokenSlug } from '../utils/token.utils';
 import { AccountTokenInterface } from './account-token.interface';
 
+export const MAINNET_TOKENS_METADATA = [
```
Comment (unixvb) [inline src/token/interfaces/account-token.interface.mock.ts:5]: is this still used?

File: src/store/wallet/wallet-reducers.ts [hunk 4]
```diff
@@ -51,6 +52,18 @@ export const walletReducers = createReducer<WalletState>(walletInitialState, bui
...
     }))
   );
+  builder.addCase(loadWhitelistAction.success, (state, { payload: tokensMetadata }) =>
+    updateCurrentAccountState(state, account => ({
+      ...account,
+      tokensList: account.tokensList.concat(
+        tokensMetadata.map(token => ({
+          slug: getTokenSlug(token),
+          balance: '0',
+          visibility: VisibilityEnum.InitiallyHidden
```
Comment (unixvb) [inline src/store/wallet/wallet-reducers.ts:62]: I saw some problems:  
1. all tokens will be hidden for empty accounts  
2. visibility will be overridden for tokens, after the app restart 


I think we need to keep some tokens visible (from the local whitelist)

File: src/navigator/main-stack.tsx [hunk 5]
```diff
@@ -76,6 +77,7 @@ export const MainStackScreen = () => {
   const initDataLoading = () => {
     dispatch(loadTezosBalanceActions.submit());
     dispatch(loadTokensActions.submit());
+    dispatch(loadWhitelistAction.submit());
```
Comment (unixvb) [inline src/navigator/main-stack.tsx:80]: I think this event should be dispatched once the app is loaded, there is no need in refetch each 30 sec

File: src/interfaces/account-state.interface.ts [hunk 6]
```diff
@@ -14,18 +14,7 @@ export interface AccountStateInterface {
...
-      balance: '0',
-      visibility: VisibilityEnum.Visible
-    })),
-    ...HIDDEN_WHITELIST_TOKENS.map(token => ({
-      slug: getTokenSlug(token),
-      balance: '0',
-      visibility: VisibilityEnum.InitiallyHidden
-    }))
-  ],
+  tokensList: [],
```
Comment (unixvb) [inline src/interfaces/account-state.interface.ts:17]: I think we still need some `INITIAL_WHITELIST_TOKENS` list  
let's discuss this on deily

**Review by unixvb — APPROVED** [review]

**Review by LendiHop — APPROVED** [review]

**Review by unixvb — APPROVED** [review]

**Review by NovakVlad2424 — APPROVED** [review]

**Pull request merged**